internal/commands/secrets/delete.go

```diff
@@ -0,0 +1,137 @@
+package secrets
+
+import (
+	"github.com/10gen/realm-cli/internal/cli"
+	"github.com/10gen/realm-cli/internal/cloud/realm"
+	"github.com/10gen/realm-cli/internal/terminal"
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/spf13/pflag"
+	"sort"
+)
+
+type CommandDelete struct {
+	inputs      deleteInputs
+	realmClient realm.Client
+	outputs     secretOutputs
+}
+
+func (cmd *CommandDelete) Inputs() cli.InputResolver {
+	return &cmd.inputs
+}
+
+func (cmd *CommandDelete) Flags(fs *pflag.FlagSet) {
+	cmd.inputs.Flags(fs)
+	fs.StringSliceVarP(&cmd.inputs.secrets, flagSecret, flagSecretShort, []string{}, flagSecretUsage)
+}
+
+func (i *deleteInputs) Resolve(profile *cli.Profile, ui terminal.UI) error {
+	if err := i.ProjectInputs.Resolve(ui, profile.WorkingDirectory); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (cmd *CommandDelete) Setup(profile *cli.Profile, ui terminal.UI) error {
+	cmd.realmClient = profile.RealmAuthClient()
+	return nil
+}
+
+func (cmd *CommandDelete) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, appErr := cli.ResolveApp(ui, cmd.realmClient, cmd.inputs.Filter())
+	if appErr != nil {
+		return appErr
+	}
+
+	secretList, secretListErr := cmd.realmClient.Secrets(app.GroupID, app.ID)
+	if secretListErr != nil {
+		return secretListErr
+	}
+
+	toDelete, resolveErr := cmd.inputs.resolveDelete(cmd.inputs.secrets, secretList, ui)
+	if resolveErr != nil {
+		return resolveErr
+	}
+
+	for _, secret := range toDelete {
+		deleteErr := cmd.realmClient.DeleteSecret(app.GroupID, app.ID, secret.ID)
+		cmd.outputs = append(cmd.outputs, secretOutput{
+			secret: secret,
+			err:    deleteErr,
+		})
+	}
+
+	return nil
+}
+
+func (cmd *CommandDelete) Feedback(profile *cli.Profile, ui terminal.UI) error {
+	if len(cmd.inputs.secrets) == 0 {
+		return ui.Print(terminal.NewTextLog("No secrets to delete"))
+	}
+
+	sort.SliceStable(cmd.outputs, secretOutputComparerBySuccess(cmd.outputs))
+	logs := terminal.NewTableLog(
+		secretDeleteMessage,
+		secretHeaders(),
+		secretTableRows(cmd.outputs, secretDeleteRow)...,
+	)
+	return ui.Print(logs)
+}
+
+func (i *deleteInputs) resolveDelete(args []string, secrets []realm.Secret, ui terminal.UI) ([]realm.Secret, error) {
```

Comment:
can we change this function signature to be:

```go
func (i *deleteInputs) resolveSecrets(ui terminal.UI, client realm.Client, app realm.App) ([]realm.Secret, error) {
  // fetch the appSecrets via the realm.Client, you probably want these put into a map[name]Secret

  // if len(i.Secrets) == 0, then we'll want to prompt the user to select (many) from the list of app secrets

  // else iterate over each i.Secret flag and figure out its corresponding ID/realm.Secret

  // return that resolved array of realm.Secret back to the command handler
}
```

basically just shifting around the bits of logic you already have.  and we shouldn't be using `args` for anything here, just rely on the flags setting values

---

also this should move to the `delete_inputs.go` file